common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -142,12 +143,53 @@ public static Geometry transform(Geometry geometry, String sourceCRS, String tar
 
     public static Geometry transform(Geometry geometry, String sourceCRS, String targetCRS, boolean lenient)
         throws FactoryException, TransformException {
-        CoordinateReferenceSystem sourceCRScode = CRS.decode(sourceCRS);
-        CoordinateReferenceSystem targetCRScode = CRS.decode(targetCRS);
+
+        CoordinateReferenceSystem sourceCRScode = parseCRSString(sourceCRS);
+        CoordinateReferenceSystem targetCRScode = parseCRSString(targetCRS);
         MathTransform transform = CRS.findMathTransform(sourceCRScode, targetCRScode, lenient);
         return JTS.transform(geometry, transform);
     }
 
+    private static CoordinateReferenceSystem parseCRSString(String CRSString) throws FactoryException{
+        if (checkCRSCodeFormat(CRSString) ){
+            return CRS.decode(CRSString);
+        }
+        else if (checkCRSWKTFormat(CRSString)){
+            return CRS.parseWKT(CRSString);
+        }
+        else {
+            return null;
+        }
+    }
+
+    private static boolean checkCRSCodeFormat(String CRSCode) {
+        try{
+            CRS.decode(CRSCode);
+        }
+        catch (FactoryException ex){
+            return false;
+        }
+
+        return true;
+    }
+
+    private static boolean checkCRSWKTFormat(String CRSWKT) {
+        try{
+            CRS.parseWKT(CRSWKT);
+        }
+        catch (FactoryException ex){
+            return false;
+        }
+
+        return true;
+    }
+
+    public static Geometry transform(Geometry geometry, CoordinateReferenceSystem sourceCRS, CoordinateReferenceSystem targetCRS, boolean lenient)
+        throws FactoryException,TransformException {
+        MathTransform transform = CRS.findMathTransform(sourceCRS,targetCRS,lenient);
+        return JTS.transform(geometry, transform);
+    }
```

Comment:
Is this used anywhere?

Reply:
Sorry for reply. 
I forgot deleting this function.
i will delete it.